java/src/test/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/DeleteCategoryControllerImplTest.java

```diff
@@ -0,0 +1,139 @@
+import static com.xxAMIDOxx.xxSTACKSxx.util.TestHelper.getBaseURL;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.HttpStatus.CONFLICT;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@EnableAutoConfiguration(
+        exclude = {
+                CosmosDbRepositoriesAutoConfiguration.class,
+                CosmosAutoConfiguration.class
+        })
+@Tag("Integration")
+class DeleteCategoryControllerImplTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private MenuRepository repository;
+
+    @AfterEach
+    void tearDown() {
+        repository.deleteAll();
+    }
+
+    @Test
+    void testDeleteCategorySuccess() {
+        // Given
+        Menu menu = createMenu(1);
+        Category category = createCategory(0);
+        menu.setCategories(List.of(category));
+        when(repository.findById(eq(menu.getId()))).thenReturn(Optional.of(menu));
+
+        // When
+        var requestEntity = getObjectHttpEntity();
+
+        String requestUrl =
+                String.format("%s/v1/menu/%s/category/%s", getBaseURL(port), menu.getId(), category.getId());
+        var response =
+                this.testRestTemplate.exchange(requestUrl, HttpMethod.DELETE, requestEntity, ErrorResponse.class);
+
+        // Then
+        ArgumentCaptor<Menu> captor = ArgumentCaptor.forClass(Menu.class);
+        verify(repository, times(1)).save(menu);
+        then(response.getStatusCode()).isEqualTo(OK);
+    }
+
+    @Test
+    void testDeleteCategoryWithAnItem() {
+        // Given
+        Menu menu = createMenu(1);
+        Category category = createCategory(0);
+        Item item =
+                new Item(UUID.randomUUID().toString(), "New Item", "New item description", 14.2d, true);
+        category.setItems(List.of(item));
+        menu.addUpdateCategory(category);
+        when(repository.findById(eq(menu.getId()))).thenReturn(Optional.of(menu));
+
+        // When
+        var requestEntity = getObjectHttpEntity();
+
+        String requestUrl =
+                String.format("%s/v1/menu/%s/category/%s", getBaseURL(port), menu.getId(), category.getId());
+        var response =
+                this.testRestTemplate.exchange(requestUrl, HttpMethod.DELETE, requestEntity, ErrorResponse.class);
+
+        // Then
+        ArgumentCaptor<Menu> captor = ArgumentCaptor.forClass(Menu.class);
+        verify(repository, times(0)).save(menu);
+        then(response.getStatusCode()).isEqualTo(CONFLICT);
+    }
+
+    private HttpEntity<Object> getObjectHttpEntity() {
```

Comment:
Strange place to put this. I'm guessing you've added the same thing elsewhere?
As mentioned, don't duplicate code